File: docs/src/main/java/jexamples/basics/pingpongtimer/Ponger.java
package jexamples.basics.pingpongtimer;

import se.sics.kompics.ComponentDefinition;
import se.sics.kompics.Negative;
import se.sics.kompics.Handler;

public class Ponger extends ComponentDefinition {

  Negative<PingPongPort> ppp = provides(PingPongPort.class);

  private long counter = 0;

  Handler<Ping> pingHandler =
      new Handler<Ping>() {
        public void handle(Ping event) {
          counter++;
          logger.info("Got Ping #{}!", counter);
          answer(event, new Pong());
        }
      };

  {
    subscribe(pingHandler, ppp);
  }
}

File: docs/src/main/scala/sexamples/basics/pingpongtimer/Ponger.scala
package sexamples.basics.pingpongtimer

import se.sics.kompics.sl._

class Ponger extends ComponentDefinition {

  val ppp = provides(PingPongPort);

  private var counter: Long = 0L;

  ppp uponEvent {
    case ping: Ping => {
      counter += 1L;
      log.info(s"Got Ping #${counter}!");
      answer(ping, Pong);
    }
  }

}
